### ISSUE TITLE
TFLite model not working with png images

### ISSUE THREAD

**Issue by UcefMountacer** [tensorflow/tensorflow#40416]
Hello,

I have trained a semantic segmentation model using VGG16 and unet. The model displayed good results before (irrelevant of image type). 

When I converted the model to tflite, I noticed that it does not work with png files. Here is an example : 

![image](https://user-images.githubusercontent.com/47783157/84525079-57424e80-acdb-11ea-9ac1-d1f1c2269b2f.png)

![image](https://user-images.githubusercontent.com/47783157/84525095-61fce380-acdb-11ea-9f92-f012c0c75ade.png)

When trying with a jpg however, the results seem fine compared to the original model. An example:

![image](https://user-images.githubusercontent.com/47783157/84525036-44c81500-acdb-11ea-9371-9015339cf568.png)

![image](https://user-images.githubusercontent.com/47783157/84525018-3e399d80-acdb-11ea-92ad-f65261f10e1a.png)

Hope someone have the answer because I do the same preprocessing to both images. And the original model as I said work fine with both types.

😃😃😃

**Comment by ElPapi42** [comment]
Can you provide a Google Colab gist reproducing the problem? can you try to compare the performance on the same image but with different encoding(.png and .jpg)? also, check out how many channels of the png images are you passing through the model? .jpg have only 3 channels but png by default have 4, maybe you trained your model only on 3-channel images and when the models perceive 4-channels, that screw everything.

**Comment by UcefMountacer** [comment]
No I did print the dimensions every time. Both are the same. Here is a code part: 

![image](https://user-images.githubusercontent.com/47783157/84556792-5cc48680-ad25-11ea-93fa-30cd19e20196.png)
![image](https://user-images.githubusercontent.com/47783157/84556799-6cdc6600-ad25-11ea-9b2d-1fbd4422fad2.png)
![image](https://user-images.githubusercontent.com/47783157/84556807-78c82800-ad25-11ea-94c3-cf2331dadd7c.png)

**Comment by amahendrakar** [comment]
@UcefMountacer,
In order to expedite the trouble-shooting process, could you please provide the `.py` file or the Python Notebook you are running along with all the supporting files. Thanks!

**Comment by amahendrakar** [comment]
Also, please mention the TensorFlow version you are using. Thanks!

**Comment by UcefMountacer** [comment]
Hi @amahendrakar,

Sorry for the late response. 

I am using the 2.2.0 version

The code I use to run the result that I have shown you is this (in TXT) : 

[main_code.txt](https://github.com/tensorflow/tensorflow/files/4808180/main_code.txt)

**Comment by srjoglekar246** [comment]
@UcefMountacer It looks like the preprocessed input in case of png file is not the same as jpg. Could you try comparing (maybe elementwise) the `image` tensors you get from a jpg vs png file after this line:

```
image = imagenet_utils.preprocess_input(image.astype(np.float32), data_format='channels_last', mode='torch')
```

I am guessing that the value of `image` elements isn't what the model was trained with, in case of png source.

**Comment by UcefMountacer** [comment]
Update : 

Sorry It took me so much time to get back on the project. I verified and there is problem not only with the tflite model, but with the original keras model. So the problem was necessarily because of preprocessing. After debugging, I found that using matplotlib to read images is not good (Yes, I don't know why I went in that direction). Using img = PIL.Image.open(path) and using np.array(img) solved the problem.

I guess this issue should be closed but another issue is risen with matplotlib.

🤙
